dm/syncer/validator_checkpoint.go

```diff
@@ -202,81 +204,49 @@ type tableChangeDataForPersist struct {
 type rowChangeDataForPersist struct {
 	Key             string           `json:"key"`
 	Tp              rowChangeJobType `json:"tp"`
+	Size            int32            `json:"size"`
 	Data            []interface{}    `json:"data"`
 	FirstValidateTS int64            `json:"first-ts"`
 	FailedCnt       int              `json:"failed-cnt"` // failed count
 }
 
-func (c *validatorPersistHelper) persist(tctx *tcontext.Context, loc binlog.Location) error {
-	// get snapshot of the current table status
-	tableStatus := c.validator.getTableStatusMap()
-	count := len(tableStatus)
-	for _, worker := range c.validator.getWorkers() {
-		for _, tblChange := range worker.getPendingChangesMap() {
-			count += len(tblChange.jobs)
-		}
-		count += len(worker.errorRows)
-	}
-	queries := make([]string, 0, count+2)
-	args := make([][]interface{}, 0, count+2)
-	nextRevision := c.revision + 1
-
-	c.L.Info("persist checkpoint and intermediate data")
-
-	// update checkpoint
-	queries = append(queries, `INSERT INTO `+c.checkpointTableName+
-		`(source, binlog_name, binlog_pos, binlog_gtid) VALUES (?, ?, ?, ?) ON DUPLICATE KEY UPDATE
-			source = VALUES(source),
-			binlog_name = VALUES(binlog_name),
-			binlog_pos = VALUES(binlog_pos),
-			binlog_gtid = VALUES(binlog_gtid)
-		`)
-	args = append(args, []interface{}{c.cfg.SourceID, loc.Position.Name, loc.Position.Pos, loc.GTIDSetStr()})
+var triggeredFailOnPersistForIntegrationTest bool
```

Comment:
What about `HasFailForIntegrationTest`...

Reply:
more verbose

Reply:
`triggeredFailOnPersistForIntegrationTest` is too long. I suggested to replace it with `HasFailForIntegrationTest`

Reply:
its too long most for the suffix ForIntegrationTest.

`HasFail` is not enough to show its meaning

Reply:
lgtm